src/modules/esl-trigger/core/esl-trigger.ts

```diff
@@ -1,84 +1,36 @@
-import {ESLBaseElement} from '../../esl-base-element/core';
 import {ExportNs} from '../../esl-utils/environment/export-ns';
-import {DeviceDetector} from '../../esl-utils/environment/device-detector';
 import {isElement} from '../../esl-utils/dom/api';
-import {setAttr} from '../../esl-utils/dom/attr';
-import {CSSClassUtils} from '../../esl-utils/dom/class';
-import {ENTER, SPACE, ESC} from '../../esl-utils/dom/keys';
-import {attr, boolAttr, prop, listen, ready} from '../../esl-utils/decorators';
-import {parseBoolean, parseNumber, toBooleanAttribute} from '../../esl-utils/misc/format';
-import {ESLMediaQuery} from '../../esl-media-query/core';
+import {attr, prop, ready} from '../../esl-utils/decorators';
 import {ESLTraversingQuery} from '../../esl-traversing-query/core';
-
 import {ESLToggleablePlaceholder} from '../../esl-toggleable/core';
 
-import type {ESLToggleable, ESLToggleableActionParams} from '../../esl-toggleable/core/esl-toggleable';
+import {ESLBaseTrigger} from './esl-base-trigger';
+
+import type {ESLToggleable} from '../../esl-toggleable/core/esl-toggleable';
+
 
 @ExportNs('Trigger')
-export class ESLTrigger extends ESLBaseElement {
+export class ESLTrigger extends ESLBaseTrigger {
   public static override is = 'esl-trigger';
   public static observedAttributes = ['target'];
 
   /** Event that represents {@link ESLTrigger} state change */
-  @prop('esl:change:active') public CHANGE_EVENT: string;
-  /** Events to observe target {@link ESLToggleable} instance state */
-  @prop('esl:show esl:hide') public OBSERVED_EVENTS: string;
-
-  /** @readonly Observed Toggleable active state marker */
-  @boolAttr({readonly: true}) public active: boolean;
-
-  /** CSS classes to set on active state */
-  @attr({defaultValue: ''}) public activeClass: string;
-  /** Target element {@link ESLTraversingQuery} selector to set `activeClass` */
-  @attr({defaultValue: ''}) public activeClassTarget: string;
+  @prop('esl:change:active') public override CHANGE_EVENT: string;
 
   /** Selector for ignored inner elements */
   @attr({defaultValue: 'a[href]'}) public ignore: string;
 
   /** Target Toggleable {@link ESLTraversingQuery} selector */
   @attr({defaultValue: ''}) public target: string;
   /** Action to pass to the Toggleable. Supports `show`, `hide` and `toggle` values. `toggle` by default */
-  @attr({defaultValue: 'toggle'}) public mode: string;
-
-  /** Click event tracking media query. Default: `all` */
-  @attr({defaultValue: 'all'}) public trackClick: string;
-  /** Hover event tracking media query. Default: `none` */
-  @attr({defaultValue: 'not all'}) public trackHover: string;
+  @attr({defaultValue: 'toggle'}) public override mode: 'toggle' | 'show' | 'hide';
```

Comment:
but as I see mode is a `@prop` on ESLBaseTrigger class

Reply:
That's right: 
 - we define the `mode` as `toggle` va `@prop` on a base level because
    - we do not want to make publically modifiable property (e.g. attribute) as we have instances of trigger that have no user cases to change the mode
    - we can not move the whole mode logic outside of the base level as it brings a lot of complications in implementing base trigger logic and produces a big breaking change in API
  - we do override the `mode` via `@attr` decorated property as it public API of the ESLTrigger (not the base instance for trigger-like instances starting from that moment)